### ISSUE TITLE
`torch.compile` silently returns a result for `torch.linalg.cholesky` on non-positive-definite input

### ISSUE THREAD

**Issue by rookieLiu2018** [pytorch/pytorch#182965]
### Describe the bug

`torch.linalg.cholesky` correctly raises an error in eager mode when the input matrix is not positive-definite. However, when the same function is compiled with `torch.compile(..., backend="inductor", fullgraph=True)`, the compiled function silently returns a tensor instead of raising an error.

### Reproducer

```python
import torch

A = torch.tensor(
    [
        [1.0, 2.0],
        [2.0, 1.0],
    ],
    dtype=torch.float32,
)

def fn(x):
    return torch.linalg.cholesky(x)

try:
    y_eager = fn(A)
    print("eager ok:")
    print(y_eager)
except Exception as e:
    print("eager error:")
    print(type(e).__name__)
    print(e)

try:
    compiled_fn = torch.compile(fn, backend="inductor", fullgraph=True)
    y_compiled = compiled_fn(A)
    print("compiled ok:")
    print(y_compiled)
except Exception as e:
    print("compiled error:")
    print(type(e).__name__)
    print(e)
```
Output:
```
eager error:
_LinAlgError
linalg.cholesky: The factorization could not be completed because the input is not positive-definite (the leading minor of order 2 is not positive-definite).
compiled ok:
tensor([[ 1.0000,  0.0000],
        [ 2.0000, -3.0000]])
```

### Versions

PyTorch version: 2.10.0+cpu

cc @malfet @jianyuh @nikitaved @mruberry @walterddr @xwang233 @Lezcano @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by RanaShiva** [comment]
Hi! I am looking to make my first contribution to PyTorch. I currently have a bug-fix PR in review over at Numba, so I am familiar with the open-source workflow and would love to help out here.

This issue looks like a great opportunity to learn about how Inductor handles error checking. My assumption is that Inductor might be omitting the check on the info flag returned by the underlying cholesky math kernel, causing it to return garbage data silently instead of raising the _LinAlgError.

Could someone assign this to me? Also, if anyone has a quick pointer to the file where torch.linalg.cholesky is lowered in the Inductor backend, I would greatly appreciate it so I can see where the error check might be missing!

**Comment by karthickai** [comment]
**🤖 PT2 Repro Agent**

I tried to reproduce this issue on PyTorch nightly (`2.13.0.dev20260514+cu130`) on NVIDIA H100 and confirmed it reproduces (3/3 runs failed).

All 3 runs reproduced the reported behavior: eager raises `_LinAlgError: linalg.cholesky: The factorization could not be completed because the input is not positive-definite (the leading minor of order 2 is not positive-definite).`, while `torch.compile(fn, backend="inductor", fullgraph=True)` silently returns `tensor([[1., 0.], [2., -3.]])` on the same non-positive-definite input. test_fn returned False on all 3 runs (exit code 1).

The compiler bisector narrowed the bug to the **aot_eager** backend; no single subsystem fully accounts for it (likely a multi-source issue accumulating across passes).

<details>
<summary>Error output</summary>

```
--- eager ---
eager error:
_LinAlgError
linalg.cholesky: The factorization could not be completed because the input is not positive-definite (the leading minor of order 2 is not positive-definite).
--- compiled ---
compiled ok:
tensor([[ 1.,  0.],
        [ 2., -3.]])

ERROR conda.cli.main_run:execute(49): `conda run python repro.py` failed. (See above for error)

```
</details>

<details>
<summary>Repro script I used</summary>

```python
import os
import torch


def test_fn() -> bool:
    """Return True if NO bug (eager and compiled both raise). False if bug present
    (compiled silently returns a tensor while eager raises _LinAlgError)."""
    torch._dynamo.reset()
    backend = os.environ.get("TORCH_COMPILE_BACKEND", "inductor")

    A = torch.tensor(
        [
            [1.0, 2.0],
            [2.0, 1.0],
        ],
        dtype=torch.float32,
    )

    def fn(x):
        return torch.linalg.cholesky(x)

    eager_raised = False
    try:
        _ = fn(A)
    except Exception:
        eager_raised = True

    compiled_raised = False
    try:
        compiled_fn = torch.compile(fn, backend=backend, fullgraph=True)
        _ = compiled_fn(A)
    except Exception:
        compiled_raised = True

    if eager_raised and not compiled_raised:
        return False
    if (not eager_raised) and compiled_raised:
        return False
    return True


if __name__ == "__main__":
    A = torch.tensor(
        [
            [1.0, 2.0],
            [2.0, 1.0],
        ],
        dtype=torch.float32,
    )

    def fn(x):
        return torch.linalg.cholesky(x)

    print("--- eager ---")
    try:
        y_eager = fn(A)
        print("eager ok:")
        print(y_eager)
    except Exception as e:
        print("eager error:")
        print(type(e).__name__)
        print(e)

    print("--- compiled ---")
    torch._dynamo.reset()
    try:
        compiled_fn = torch.compile(fn, backend="inductor", fullgraph=True)
        y_compiled = compiled_fn(A)
        print("compiled ok:")
        print(y_compiled)
    except Exception as e:
        print("compiled error:")
        print(type(e).__name__)
        print(e)

    try:
        ok = test_fn()
    except Exception:
        import traceback
        traceback.print_exc()
        os._exit(2)
    os._exit(0 if ok else 1)

```
</details>

<details>
<summary>My environment</summary>

```
Collecting environment information...
PyTorch version: 2.13.0.dev20260514+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: CentOS Stream 9 (x86_64)
GCC version: (GCC) 11.5.0 20240719 (Red Hat 11.5.0-14)
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.34

Python version: 3.13.12 | packaged by Anaconda, Inc. | (main, Feb 24 2026, 16:13:31) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.16.1-0_fbk2_hardened_0_gf40efc324cc8-x86_64-with-glibc2.34
Is CUDA available: True
CUDA runtime version: 12.8.93
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: NVIDIA H100
Nvidia driver version: 580.126.09
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           52 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  46
On-line CPU(s) list:                     0-45
Vendor ID:                               AuthenticAMD
Model name:                              AMD EPYC 9654 96-Core Processor
CPU family:                              25
Model:                                   17
Thread(s) per core:                      1
Core(s) per socket:                      46
Socket(s):                               1
Stepping:                                1
BogoMIPS:                                4792.79
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm rep_good nopl cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy svm cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw perfctr_core ssbd ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves avx512_bf16 clzero xsaveerptr wbnoinvd arat npt lbrv nrip_save tsc_scale vmcb_clean flushbyasid pausefilter pfthreshold v_vmsave_vmload vgif vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid fsrm flush_l1d arch_capabilities
Virtualization:                          AMD-V
Hypervisor vendor:                       KVM
Virtualization type:                     full
L1d cache:                               2.9 MiB (46 instances)
L1i cache:                               2.9 MiB (46 instances)
L2 cache:                                23 MiB (46 instances)
L3 cache:                                736 MiB (46 instances)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-45
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Old microcode:             Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Vulnerable
Vulnerability Spec store bypass:         Vulnerable
Vulnerability Spectre v1:                Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
Vulnerability Spectre v2:                Vulnerable; IBPB: disabled; STIBP: disabled; PBRSB-eIBRS: Not affected; BHI: Not affected
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Vulnerable
Vulnerability Tsx async abort:           Not affected

Versions of relevant libraries:
[pip3] mypy_extensions==1.1.0
[pip3] numpy==2.4.4
[pip3] nvidia-cublas==13.1.1.3
[pip3] nvidia-cuda-cupti==13.0.85
[pip3] nvidia-cuda-nvrtc==13.0.88
[pip3] nvidia-cuda-runtime==13.0.96
[pip3] nvidia-cudnn-cu13==9.20.0.48
[pip3] nvidia-cufft==12.0.0.61
[pip3] nvidia-curand==10.4.0.35
[pip3] nvidia-cusolver==12.0.4.66
[pip3] nvidia-cusparse==12.6.3.3
[pip3] nvidia-cusparselt-cu13==0.8.1
[pip3] nvidia-nccl-cu13==2.29.7
[pip3] nvidia-nvjitlink==13.0.88
[pip3] nvidia-nvtx==13.0.85
[pip3] torch==2.13.0.dev20260514+cu130
[pip3] torchao==0.17.0
[pip3] torchaudio==2.11.0.dev20260515+cu130
[pip3] torchdata==0.11.0
[pip3] torchmetrics==1.0.3
[pip3] torchrec==1.7.0.dev20260413+cu130
[pip3] torchtune==0.6.1
[pip3] torchvision==0.28.0.dev20260515+cu130
[pip3] triton==3.7.0+git88b227e2
[conda] Could not collect
```
</details>

---
If the repro agent produced an incorrect result (wrong classification, bad repro script, missed the actual bug, etc.), add the label `repro:agent:h100:incorrect` to the issue.

**Comment by morrison-turnansky** [comment]
looking into this

**Comment by morrison-turnansky** [comment]
my pr should fix this, but it looks like https://github.com/pytorch/pytorch/pull/184111 will fix it as well. I will circle back after that one gets merged.

**Comment by jansel** [comment]
My agent was unable to reproduce this on 40e21dcd4b92d59842b3e3b7f542f855dedddb91. Reopen if you see this on the latest PyTorch nightly
